src/main/java/laddergame/controller/LadderController.java

```diff
@@ -31,19 +34,68 @@ public LadderController(final InputView inputView,
     }
 
     public void run() {
-        final Participants participants = retryOnRuntimeExceptionWithMessage(() -> new Participants(readNames()));
-        final Width ladderWidth = new Width(participants.getSize());
-        final Height ladderHeight = retryOnRuntimeExceptionWithMessage(() -> new Height(inputView.readHeight()));
-        final Ladder ladder = new Ladder(ladderWidth, ladderHeight, booleanGenerator);
+        final PersonalNames names = retryOnInvalidInputWithAlert(
+                () -> new PersonalNames(inputView.readNames()));
+        final LadderResult ladderResult = retryOnInvalidInputWithAlert(
+                () -> LadderResult.of(names, inputView.readResults()));
+        final Height ladderHeight = retryOnInvalidInputWithAlert(
+                () -> new Height(inputView.readHeight()));
+        final Ladder ladder = new Ladder(new Width(names.getSize()), ladderHeight, booleanGenerator);
+
+        showGeneratedLadder(names, ladderResult, ladder);
+        playGame(names, ladderResult, ladder);
+    }
```

Comment:
```suggestion
        final PersonalNames names = createPersonalNames(inputView.readNames());
        final LadderResult ladderResult = createLadderResult(names, inputView.readResults());
        final Height ladderHeight = createHeight(inputView.readHeight());

        final Ladder ladder = new Ladder(new Width(names.getSize()), ladderHeight, booleanGenerator);

        showGeneratedLadder(names, ladderResult, ladder);
        playGame(names, ladderResult, ladder);
    }

    private PersonalNames createPersonalNames(List<String> names) {
        return createInstance(() -> new PersonalNames(names));
    }

    private LadderResult createLadderResult(PersonalNames personalNames, List<String> results) {
        return createInstance(() -> LadderResult.of(personalNames,results));
    }

    private Height createHeight(int height) {
        return createInstance(() -> new Height(height));
    }
```

는 어떠세요?
그럼에도, 중복코드는 어쩔수없네요.

Reply:
좋습니다! 훨신 깔끔해진 것 같아요~~